ToolBox/SamcoDataDownloader/SamcoDataDownloader.cs

```diff
@@ -0,0 +1,101 @@
+        /// </summary>
+        public static void SamcoDataDownloader(IList<string> tickers, string market, string resolution, string securityType, DateTime startDate, DateTime endDate)
+        {
+            if (resolution.IsNullOrEmpty() || tickers.IsNullOrEmpty())
+            {
+                Log.Error("SamcoDataDownloader ERROR: '--tickers=', --securityType, '--market' or '--resolution=' parameter is missing");
+                Log.Error("--tickers=eg JSWSTEEL,TCS,INFY");
+                Log.Error("--market=MCX/NSE/NFO/CDS/BSE");
+                Log.Error("--security-type=Equity/Future/Option");
+                Log.Error("--resolution=Minute/Hour/Daily/Tick");
+                Environment.Exit(1);
+            }
+            try
+            {
+                var _samcoAPI = new SamcoBrokerageAPI();
+                _samcoAPI.Authorize(Config.Get("samco-client-id"), Config.Get("samco-client-password"), Config.Get("samco-year-of-birth"));
+
+                var castResolution = (Resolution)Enum.Parse(typeof(Resolution), resolution);
+                var castSecurityType = (SecurityType)Enum.Parse(typeof(SecurityType), securityType);
+
+                if (castSecurityType == SecurityType.Forex || castSecurityType == SecurityType.Cfd || castSecurityType == SecurityType.Crypto || castSecurityType == SecurityType.Base)
+                {
+                    throw new ArgumentException("Invalid security type: " + castSecurityType);
+                }
+
+                if (startDate >= endDate)
+                {
+                    throw new ArgumentException("Invalid date range specified");
+                }
+
+                // Load settings from config.json and create downloader
+                var dataDirectory = Globals.DataFolder;
+                var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+                var symbolMapper = new SamcoSymbolMapper();
+
+                foreach (var pair in tickers)
+                {
+                    // Download data
+                    var pairObject = Symbol.Create(pair, castSecurityType, market);
+                    var exchange = symbolMapper.GetDefaultExchange(pairObject);
+                    var scrip = symbolMapper.SamcoSymbols.Where(x => x.Name.ToUpperInvariant() == pairObject.ID.Symbol).First();
+                    var isIndex = scrip.Instrument == "INDEX";
+
+                    // Write data
+                    var writer = new LeanDataWriter(castResolution, pairObject, dataDirectory);
+                    IList<TradeBar> fileEnum = new List<TradeBar>();
+                    var history = new List<TradeBar>();
+                    if (castResolution == Resolution.Tick)
+                    {
+                        throw new ArgumentException("Samco Doesn't support tick resolution");
+                    }
+
+                    history = _samcoAPI.GetIntradayCandles(pairObject, exchange, startDate, endDate, isIndex: isIndex).ToList();
```

Comment:
nit- don't need to call `ToList` here if going to iterate again bellow in the foreach